[![Docs](https://img.shields.io/badge/docs-latest-blue.svg)](https://osf.io/t6j7u/wiki/home/) 
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)

# Building Mutational Signatures in Cancer using Deep Bayesian Neural Nets

<sub><sub>**J.A. Busker**</sub></sub>    
<sub><sub>**Hanze University of Applied Sciences, Groningen**</sub></sub>    
<sub><sub>**UMCG department of Epidemiology, Groningen**</sub></sub>

This project aims to refine the statistical model and the current representation of mutations in building mutational signatures in cancer using deep Bayesian neural nets. Cancer is characterized by uncontrolled growth of cells, which is primarily acquired by mutation of the genome. These mutations have accrued by exposure to DNA-damaging processes coming from within (endogenous) and from outside (exogenous) throughout the course of life. By looking at the direct context of the mutations, we can infer the way these mutations were created (aetiology).
	
## Table of Contents

- [Introduction](#introduction)
- [Project Structure](#project-structure)
- [Installation](#installation)
- [Usage](#usage)

## Introduction

Cancer is a disease of the genome, characterized by uncontrolled growth of cells acquired by mutation of the genome. These mutations have accrued by exposure to DNA-damaging processes coming from within and outside throughout the course of life. By looking at the direct context of the mutations, we can infer the way these mutations were created (aetiology). Most researchers have looked at the changing mutation including a single letter of context to the left and right. This has resulted in a curated set of mutational signatures by COSMIC, which can be viewed online. In addition, several methods have been developed to determine the signatures and infer the attributions. The goal of this project is to refine the statistical model and the current representation of the mutations.

## Project Structure

The project is organized into the following folders:

### `data`

The `data` folder is intended for storing datasets or data files used in the project.

### `documents`

The `documents` folder contains important documents related to school, research, or any relevant materials. These documents may include academic papers, project proposals, and other resources that contribute to the understanding and development of the project.

### `notebook`

The `notebook` folder is intended for storing notebooks for storing finds that were found.

### `results`

The `results` folder is designated for storing output files, plots, or any results generated by the code.

### `scripts`

The `scripts` contains executable scripts that perform specific tasks or experiments within the project.

## Installation

1. Clone the repository & Install dependencies:

```bash
$ git clone https://github.com/AlfonsoJan/DeepBayesMutSig
```

2. Set up a virtual environment (optional, but recommended):

```bash
$ pip install virtualenv
$ virtualenv venv
```

3. Activate virtual environment:

On Windows:
	
```bash
$ venv/Scripts/activate
```

On macOS/Linux:

```bash
$ source venv/bin/activat
```

4. Install dependencies:

```bash
$ pip install -r requirements.txt
```

## Usage

The scripts are meant to be run in order.

### Setup

Install your desired reference genome as follows:

```python
from GenomeSigInfer.utils import download
ref_genome = "project/ref_genome" # Folder where the ref genome will be downloaded
genome = "GRCh37" # Reference Genome
bash = False # If you want to download the ref genome using bash
download.download_ref_genome(ref_genome, genome, bash)
```

### Create SBS Files

Create mutliple SBS files. With increasing context.

The sbs.96.txt file contains all of the following the pyrimidine single nucleotide variants, N[{C > A, G, or T} or {T > A, G, or C}]N.
*4 possible starting nucleotides x 6 pyrimidine variants x 4 ending nucleotides = 96 total combinations.*

The sbs.1536.txt file contains all of the following the pyrimidine single nucleotide variants, NN[{C > A, G, or T} or {T > A, G, or C}]NN.
*16 (4x4) possible starting nucleotides x 6 pyrimidine variants x 16 (4x4) possible ending nucleotides = 1536 total combinations.*

The sbs.24576.txt file contains all of the following the pyrimidine single nucleotide variants, NNN[{C > A, G, or T} or {T > A, G, or C}]NNN.
*16 (4x4) possible starting nucleotides x 16 (4x4) nucleotides x 6 pyrimidine variants x 16 (4x4) nucleotides x 16 (4x4) possible ending dinucleotides = 24576 total combinations.*

```python
from GenomeSigInfer.sbs import SBSMatrixGenerator
sbs_out = "project/SBS" # Where the SBS files will be saved to
ref_genome = Path("project/ref_genome") # Parent folder of the Ref genome
vcf = (Path("test/files/test.vcf"),) # Tuple of the VCF file(s)
genome = "GRCh37" # Ref genome
SBSMatrixGenerator.generate_sbs_matrix(
    folder=sbs_out, ref_genome=ref_genome, vcf_files=vcf, genome=genome
)
```

or run `scripts\create_sbs_files.py` file.

```bash
$ python scripts/create_sbs_files.py
```

### Extract Signatures from SBS files

Create nmf files and deconmpose mutational signatures from NMF results. And calculate the similarities between each file's signature data and the cosmic columns.

```python
from GenomeSigInfer.nmf import NMFMatrixGenerator
sbs_dir = "project/SBS"  # Folder where the SBS files are located
nmf_dir = "project/NMF" # Folder where the NMF files are saved to
sigs = 48 # Amount of mutational signatures
cosmic = "/path/to/cosmic.file" # Cosmic file name for the results (Path Object)
res_dir = "project/results" # Folder where results from the analysis will be saved to
nmf_init = "nndsvda" # Put here the best init from the last step
beta_loss = "frobenius" # Put here the best beta_loss from the last step
NMFMatrixGenerator.generate_nmf_matrix(
    sbs_folder=sbs_dir,
    signatures=sigs,
    cosmic=cosmic,
    nmf_folder=nmf_dir,
    nmf_init=nmf_init,
    beta_los=beta_loss,
    result_folder=res_dir,
)
```

or run `scripts\create_nmf_files.py` file.

```bash
$ python scripts/create_nmf_files.py
```

### Create Signature Plots

Create signature plots for all the decomposed signatures files.

```python
from GenomeSigInfer.figures import signature_plots
nmf_folder = "project/NMF" # Folder where the NMF files are located
result_folder = "project/results" # Folder where the plots are saved to
sig_plots = signature_plots.SigPlots(nmf_folder, result_folder)
sig_plots.create_plots()
sig_plots.create_expected_plots()
```

or run `scripts\create_signatures_plots.py` file.

```bash
$ python scripts/create_signatures_plots.py
```

## References

- Alexandrov, Ludmil B., et al. "The repertoire of mutational signatures in human cancer." Nature 578.7793 (2020): 94-101.
- Bergstrom, Erik N., et al. "SigProfilerMatrixGenerator: a tool for visualizing and exploring patterns of small mutational events." BMC genomics 20.1 (2019): 1-12.
- Degasperi, Andrea, et al. "Substitution mutational signatures in whole-genomeâ€“sequenced cancers in the UK population." Science 376.6591 (2022): abl9283.
- Kim, Yoo-Ah, et al. "Mutational signatures: From methods to mechanisms." Annual Review of Biomedical Data Science 4 (2021): 189-206.
- Zhou, Mingyuan, Yulai Cong, and Bo Chen. "Augmentable gamma belief networks." The Journal of Machine Learning Research 17.1 (2016): 5656-5699.
