{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from GenomeSigInfer.figures import signature_plots\n",
    "from pathlib import Path\n",
    "nmf_folder = Path(\"../NMF\") # Folder where the NMF files are located\n",
    "figure_folder = Path(\"../figures\") # Folder where the plots are saved to\n",
    "sig_plots = signature_plots.SigPlots(nmf_folder, figure_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:26:45 - INFO - Creating siganture plots for context: '96'\n",
      "2024-01-19 14:27:05 - INFO - Creating siganture plots for context: '1536'\n",
      "2024-01-19 14:28:13 - INFO - Creating siganture plots for context: '24576'\n"
     ]
    }
   ],
   "source": [
    "sig_plots.create_plots()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SBS2 / SBS13\n",
    "\n",
    "Are associated with the activation of AID/APOBEC cytidine deaminases in cancer, possibly due to previous viral infection, retrotransposon jumping, or tissue inflammation.\n",
    "\n",
    "SBS13 mutations are likely generated by error-prone polymerases replicating across abasic sites generated by base excision repair removal of uracil [1]. All A3 and AID enzymes deaminate the second cytidine in either CC or TC dinucleotide motifs [2]. However, sequence analysis has determined that not every cognate dinucleotide motif is deaminated, which suggests that the nucleotides flanking the dinucleotide motifs play a role in substrate recognition [2]. The pattern in SBS2 escpecially ion the context of 7, where the majority of the nucleotiudes are thymine, could be explained by the binding residue of A3Fâ€CTD poly T ssDNA [2].\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:30:42 - INFO - Creating Signature plot for SBS2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:30:54 - INFO - Created Signature plot for SBS2\n",
      "2024-01-19 14:30:54 - INFO - Creating Signature plot for SBS13\n",
      "2024-01-19 14:30:59 - INFO - Created Signature plot for SBS13\n"
     ]
    }
   ],
   "source": [
    "sig_plots.create_expected_plots([\"SBS2\", \"SBS13\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1 = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7370881/\\\n",
    "2 = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6699113/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SBS7a\n",
    "\n",
    "SBS7a signature, which is associated with skin cancers from sun-exposed areas and is likely due to exposure to ultraviolet light [4]. This signature is characterized C>T mutations and TT dinucleotides [3]. In both context the most dominant nucleotide is thymine."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:30:59 - INFO - Creating Signature plot for SBS7a\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:31:04 - INFO - Created Signature plot for SBS7a\n"
     ]
    }
   ],
   "source": [
    "sig_plots.create_expected_plots([\"SBS7a\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3 = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9402440/\\\n",
    "4 = https://onlinelibrary.wiley.com/doi/full/10.1002/ijc.34402"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SBS10a\n",
    "SBS10a is a mutational signature that is often associated with large numbers of somatic mutations, leading to samples with these signatures being termed hypermutators. The etiology of SBS10a is possibly driven by the POLE (polymerase epsilon) exonuclease domain mutations [1,5]. SBS10a split from SBS7 [7], SBS10a is characterized by C>A mutation and TT dinucleotides."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:31:04 - INFO - Creating Signature plot for SBS10a\n",
      "2024-01-19 14:31:09 - INFO - Created Signature plot for SBS10a\n"
     ]
    }
   ],
   "source": [
    "sig_plots.create_expected_plots([\"SBS10a\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "5 = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9266985/\\\n",
    "6 = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4874836/\\\n",
    "7 = https://www.nature.com/articles/s41586-020-1943-3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SBS17b\n",
    "\n",
    "SBS17b is a mutational signature that has been associated with fluorouracil (5FU) chemotherapy treatment and damage inflicted by reactive oxygen species [8]. SBS17b is considered an easily trackable signature and it has a marked characteristic of T>G substitutions, possibly caused by oxidative damage [9]. ALso split from SBS7 [7].\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:31:09 - INFO - Creating Signature plot for SBS17b\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:31:14 - INFO - Created Signature plot for SBS17b\n",
      "2024-01-19 14:31:14 - INFO - Creating Signature plot for SBS17a\n",
      "2024-01-19 14:31:18 - INFO - Created Signature plot for SBS17a\n"
     ]
    }
   ],
   "source": [
    "sig_plots.create_expected_plots([\"SBS17b\"])\n",
    "sig_plots.create_expected_plots([\"SBS17a\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "8 = https://cancer.sanger.ac.uk/signatures/sbs/sbs17b/\\\n",
    "9 = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9353727/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SBS22a\n",
    "\n",
    "SBS22a is a mutational signature associated with exposure to aristolochic acid (AA), a compound found in certain plants and herbal remedies that has been linked to cancer [8]. Is a signatures characterized by T>A [10]. Aristolochic-acid exposure and similar patterns SBS22 is due to aristolochic acid [10]. The mutational signature of AA-induced DNA damage is characterized by a predominance of A:T-to-T:A transversions, a relatively unusual type of mutation that is infrequently seen in other types of cancer, including those caused by other carcinogens. These mutations concentrate at splice sites, causing the inappropriate inclusion or exclusion of entire exons in the resulting mRNA [11].\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:31:18 - INFO - Creating Signature plot for SBS22a\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-01-19 14:31:24 - INFO - Created Signature plot for SBS22a\n"
     ]
    }
   ],
   "source": [
    "sig_plots.create_expected_plots([\"SBS22a\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "10 = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7613262/\\\n",
    "11 = https://www.science.org/doi/full/10.1126/scitranslmed.3006086"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
